src/test/java/com/epam/mjc/io/FileReaderTest.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.mjc.io;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class FileReaderTest {
+
+    private final static Path TEST_FILE = Path.of("src/main/resources/Profile.txt");
+    private final static Path READER_CLASS = Path.of("src/main/java/com/epam/mjc/io/FileReader.java");
+    private static final String PROFILE_PARAM_VALUE_DELIMETER = ": ";
+    private static final String SOURCE_CODE_LOCATION = "src/main/java";
+
+
+    @Test
+    public void testProgramReadsDataCorrectly () {
+        FileReader fileReader = new FileReader();
+        Profile actual = fileReader.getDataFromFile(new File(TEST_FILE.toUri()));
+        Profile expected = readFile(TEST_FILE);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testCodeWithoutExternalUtils() throws IOException {
+        final Path sources = Path.of(SOURCE_CODE_LOCATION);
+        Files.walk(sources)
+                .filter(Files::isRegularFile)
+                .filter(p -> p.toString().endsWith(".java"))
+                .forEach(this::assertSourceWithoutProhibitedLibraries);
+    }
+
+    @Test
+    public void testCodeHasStreamClosing() {
+        assertSourceHasClosing(READER_CLASS);
+    }
+
+    private void assertSourceHasClosing(Path path) {
+        String sourceCode = readFileIntoString(path);
+        assertTrue(sourceCode.contains("try (") || sourceCode.contains("try(") || sourceCode.contains(".close()"));
+    }
+
+    private void assertSourceWithoutProhibitedLibraries(Path path) {
+        String sourceCode = readFileIntoString(path);
+        assertFalse(sourceCode.contains(".nio"));
+        assertFalse(sourceCode.contains("Files"));
```

Comment:
Do you really need check `Files`? As I can see, `Files` is class from `nio` package.

Reply:
I agree

Reply:
Fixed